pubsub/pulsar/pulsar.go

```diff
@@ -137,11 +139,17 @@ func parsePulsarMetadata(meta pubsub.Metadata) (*pulsarMetadata, error) {
 				value:    v,
 			}
 		} else if strings.HasSuffix(k, topicAvroSchemaIdentifier) {
-			topic := k[:len(k)-len(topicJSONSchemaIdentifier)]
+			topic := k[:len(k)-len(topicAvroSchemaIdentifier)]
 			m.internalTopicSchemas[topic] = schemaMetadata{
 				protocol: avroProtocol,
 				value:    v,
 			}
+		} else if strings.HasSuffix(k, topicProtoSchemaIdentifier) {
```

Comment:
Total nit, but since we have 3 cases now it may be nice to change this to a switch

Reply:
Changed